src/user/user.service.spec.ts

```diff
@@ -0,0 +1,151 @@
+import { Test, TestingModule } from "@nestjs/testing";
+import { getRepositoryToken } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { CreateUserDto } from "./dto/create-user.dto";
+import { UpdateUserDto } from "./dto/update-user.dto";
+import { User } from "./user.entity";
+import { UserService } from "./user.service";
+
+const mockUserData: User[] = [
+  {
+    id: 1,
+    auth_id: "auth0|6237590b647a36006ba3c6ea",
+    username: "johnsmith",
+    created: new Date(2022, 2, 30, 15, 32),
+    modified: new Date(2022, 3, 1, 24, 55),
+  },
+  {
+    id: 2,
+    auth_id: "auth0|6237590b6aaaaajfsdoalij2",
+    username: "testuser",
+    created: new Date(2022, 3, 1, 20, 44),
+    modified: new Date(2022, 3, 1, 20, 44),
+  },
+];
+
+describe("UserService", () => {
+  let service: UserService;
+  let repository: Repository<User>;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        UserService,
+        {
+          provide: getRepositoryToken(User),
+          useValue: {
+            find: jest.fn().mockResolvedValue(mockUserData),
+            findOne: jest.fn().mockResolvedValue(mockUserData[0]),
+            save: jest.fn().mockResolvedValue(mockUserData[0]),
+          },
+        },
+      ],
+    }).compile();
+
+    service = module.get<UserService>(UserService);
+    repository = module.get<Repository<User>>(getRepositoryToken(User));
+  });
+
+  it("should be defined", () => {
+    expect(service).toBeDefined();
+    expect(repository).toBeDefined();
+  });
+
+  describe("UserService#getAll", () => {
```

Comment:
I don't think there is any benefit as far as I can tell you putting the class and method name in this, since the class name is in the parent `describe()` block. I would probably just do:
```ts
describe("ClassName", () => {
    describe("methodName()", () => {
        it("has a test case...", () => {});
    });
});
```

Reply:
Check